bugwarrior/services/gmail.py

```diff
@@ -178,12 +179,27 @@ def get_labels(self):
 
     def get_threads(self):
         thread_service = self.gmail_api.users().threads()
+        threads = []
 
-        result = thread_service.list(
-            userId=self.config.login_name, q=self.config.query).execute()
-        return [
-            thread_service.get(userId='me', id=thread['id']).execute()
-            for thread in result.get('threads', [])]
+        pageToken = None
+
+        while len(threads) < self.config.thread_limit:
+            maxResults = min(100, self.config.thread_limit - len(threads))
+
+            result = thread_service.list(userId=self.config.login_name, q=self.config.query,
+                                         maxResults=maxResults, pageToken=pageToken).execute()
+
+            for thread in result.get('threads', []):
+                threads.append(thread_service.get(userId='me', id=thread['id']).execute())
+
+            if len(result.get('threads', [])) == 0:
+                break
```

Comment:
Is this `break` condition necessary? Surely in any circumstance in which no threads are returned there would also be no `nextPageToken` and the next `break` condition would apply.

Reply:
checking it even after for loop does not make sense